llvm/test/Transforms/InstCombine/fcmp.ll

```diff
@@ -596,8 +596,8 @@ define i1 @is_signbit_set(double %x) {
 
 define i1 @is_signbit_set_1(double %x) {
 ; CHECK-LABEL: @is_signbit_set_1(
-; CHECK-NEXT:    [[S:%.*]] = call double @llvm.copysign.f64(double 1.000000e+00, double [[X:%.*]])
-; CHECK-NEXT:    [[R:%.*]] = fcmp ult double [[S]], 0.000000e+00
+; CHECK-NEXT:    [[TMP1:%.*]] = bitcast double [[X:%.*]] to i64
+; CHECK-NEXT:    [[R:%.*]] = icmp slt i64 [[TMP1]], 0
```

Comment:
Given that we currently don't try to track FP class / sign bit through bitcast, this form seems worse? Plus, if we had fast math flags this would necessarily lose them

Reply:
Should we introduce a new intrinsic `i1 @llvm.signbit`?

Reply:
If it returns an i1, that wouldn't really be any better. It still wouldn't be a candidate for tracking. If we returned the signbit encoded in the original FP type, it may be more useful